presto-main/src/main/java/com/facebook/presto/spiller/FileSingleStreamSpillerFactory.java

```diff
@@ -154,6 +154,9 @@ private synchronized Path getNextSpillPath()
                 return path;
             }
         }
+        if (spillPaths.isEmpty()) {
```

Comment:
I was actually thinking about adding this check on Presto startup. So we check if spill paths are configured if spill is enabled. I am not sure if there is natural place for such config cross validation. If there is no better place you could pass `spill.enabled` flag to `FileSingleStreamSpillerFactory` and put validation in `@PostConstruct` method.
What do you think?

Reply:
That would work. However, you can enable/disable spill on session level too. Indeed, Presto can refuse to start when spill is enabled and no paths configured, but it still must fail reasonably when no paths configured and users enables spill with a session's switch.

Reply:
Actually there is no session property to change spill paths.
And there is already code which validates if spill paths list is non-empty if we enable spilling via session property 

```
                new PropertyMetadata<>(
                        SPILL_ENABLED,
                        "Experimental: Enable spilling",
                        BOOLEAN,
                        Boolean.class,
                        featuresConfig.isSpillEnabled(),
                        false,
                        value -> {
                            boolean spillEnabled = (Boolean) value;
                            if (spillEnabled && featuresConfig.getSpillerSpillPaths().isEmpty()) {
                                throw new PrestoException(
                                        StandardErrorCode.INVALID_SESSION_PROPERTY,
                                        format("%s cannot be set to true; no spill paths configured", SPILL_ENABLED));
                            }
                            return spillEnabled;
                        },
                        value -> value),
```

Hmm this is awful to have it in two places. 

Maybe instead adding special validation for `spill-paths` config property we should validate if default session is valid on Presto startup?

Reply:
Default session validation would produce the wrong error message, since the problem is the spill config property, not the session property. I think we need the check in both places. But I agree we should do it on startup if spilling is enabled by default.